### PR TITLE
[test-only][full-ci] Given step implementation of uploading various types of files in the server

### PR DESCRIPTION
**Pull request by ishabaral**
## Description
Implemented `on the server` steps in the local test directory.
This `Given` step implementation uploads various types of files in the server

Renamed step:

```diff
-Given user <user> has uploaded file <fileName> to "<fileName>" on the server
+Given user <user> has uploaded file <fileName> to "<fileName>" in the server
```
Part of https://github.com/owncloud/client/issues/10432

### CODE REVIEW

File: test/gui/shared/scripts/helpers/ConfigHelper.py [hunk 1]
```diff
@@ -176,3 +180,8 @@ def set_config(key, value):
 def clear_scenario_config():
     for key, value in SCENARIO_CONFIGS.items():
         CONFIG[key] = value
+
+
+def get_local_upload_dir(file_name):
+    base_upload_dir = get_config('files_for_upload')
+    return os.path.join(base_upload_dir, file_name)
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/ConfigHelper.py:187]: this should not be in the config

File: test/gui/shared/scripts/helpers/ConfigHelper.py [hunk 2]
```diff
@@ -100,6 +103,7 @@ def get_default_home_dir():
     'record_video_on_failure': False,
     'retrying': False,
     'video_recording_started': False,
+    'files_for_upload': CURRENT_DIR.parent.parent / 'upload-files',
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/ConfigHelper.py:106]:
```suggestion
    'files_for_upload': CURRENT_DIR.parent.parent / 'files-for-upload',
```

Comment (saw-jan) [inline test/gui/shared/scripts/helpers/ConfigHelper.py:106]:
```suggestion
    'files_for_upload': os.path.join(CURRENT_DIR.parent.parent, 'upload-files'),
```

Comment (saw-jan) [inline test/gui/shared/scripts/helpers/ConfigHelper.py:106]: is this correct `parent.parent.parent`?
```suggestion
    'files_for_upload': CURRENT_DIR / 'upload-files',
```

File: test/gui/shared/scripts/helpers/ConfigHelper.py [hunk 3]
```diff
@@ -176,3 +180,8 @@ def set_config(key, value):
 def clear_scenario_config():
     for key, value in SCENARIO_CONFIGS.items():
         CONFIG[key] = value
+
+
+def get_local_upload_dir(file_name):
```
Comment (saw-jan) [inline test/gui/shared/scripts/helpers/ConfigHelper.py:185]: ? :thinking: 
```suggestion
def get_file_for_upload(file_name):
```

File: test/gui/shared/scripts/helpers/ConfigHelper.py (hunk 1 above)
Comment (saw-jan) [reply test/gui/shared/scripts/helpers/ConfigHelper.py:187]: `file_helper` maybe  the right place

File: test/gui/shared/scripts/helpers/ConfigHelper.py (hunk 2 above)
Comment (ishabaral) [reply test/gui/shared/scripts/helpers/ConfigHelper.py:106]: Used `parent.parent` to move up two levels from the `CURRENT_DIR`

**Review by saw-jan — APPROVED** [review]
:+1: :sparkles:

**Pull request merged**